sympy/assumptions/handlers/matrices.py

```diff
@@ -38,6 +38,8 @@ def MatMul(expr, assumptions):
         factor, mmul = expr.as_coeff_mmul()
         if all(ask(Q.symmetric(arg), assumptions) for arg in mmul.args):
             return True
+        if ask(Q.diagonal(expr), assumptions):
```

Comment:
I've added TODO comments in SymmetricHandler. New sathandlers are out of a scope of this pull request. I'll investigate if I can implement them. If so I'll send a new pull request.